runtime/compiler/net/CommunicationStream.cpp

```diff
@@ -73,6 +73,20 @@ CommunicationStream::readMessage(Message &msg)
 
    // rebuild the message
    msg.deserialize();
+
+   // collect message size
+/*#ifdef STATS
+   static TR_Stats statMsgSize[MessageType_ARRAYSIZE];
+   statMsgSize[int(msg.type())].update(messageSize);
+   statMsgSize[int(msg.type())].report(stderr);
+#endif*/
+   TR::CompilationInfo *compInfo;
+   static TR_Stats collectMsgStat[MessageType_ARRAYSIZE];
+   collectMsgStat[int(msg.type())].update(messageSize); //I dont need it anymore since I pass the type and size
+   //compInfo->setStatMsgSize(collectMsgStat);
+   compInfo->setMsgSize(int(msg.type()), messageSize);
```

Comment:
compInfo is not initialized here. This could lead to crashes

Reply:
Since you don't have easy access to compInfo from communicationStream it may be better to define that `TR_Stats collectMsgStat[MessageType_ARRAYSIZE];` as a static field in CommunicationStream class.